spec/spec_helper.rb

```diff
@@ -39,6 +39,6 @@
   config.append_after(:each, db: true) do |ex|
     ActiveRecord::Base.connection.rollback_transaction
 
-    raise "Migrations are pending: #{ex.metadata[:location]}" if ActiveRecord::Base.connection.migration_context.needs_migration?
+    raise "Migrations are pending: #{ex.metadata[:location]}" if ActiveRecord::Base.connection_pool.migration_context.needs_migration?
```

Comment:
Looks like this change is causing test failures on CI. Could you please take a look?

Reply:
Will do. This does pass locally with 7.2.2 and Ruby 3.3.6. Additionally, this isn't a backward compatible change so I'll fix that too.

Reply:
Still investigating this piece.

Reply:
The last two commits should address this. See https://github.com/atomaka/logidze/actions/runs/12209635303/job/34064707411 (from https://github.com/atomaka/logidze/pull/2) for a passing test suite.